### ISSUE TITLE
Revise searches and documentation top-level

### ISSUE THREAD

**Issue by jeapostrophe** [racket/racket#1797]

**Comment by jeapostrophe** [comment]
@mflatt Can you look through my implementation notes?

**Comment by jackfirth** [comment]
This also might be easier if the package catalog served both HTML and s-exp data for packages from the same URL using content negotiation. There wouldn't need to be any sort of URL templating enforced on the package catalog, it'd just have to respond differently based on the Accept header.

**Comment by mflatt** [comment]
@jeapostrophe Looks generally good. I had forgotten about `current-pkg-catalog-file`, and it seems like a fine local catalog to use by default and update on demand; that's the same one used by the GUI package manager. As you and I discussed, I think we'll need a new configuration (as set by `raco pkg config` and reported via `setup/dirs`) to configure a target from links from package names.

@jackfirth If I understand what you mean, that's difficult or impossible with the current S3-backed setup.

**Comment by jackfirth** [comment]
@mflatt Yes, the current setup makes this difficult. An alternative setup using a pull-based model with AWS CloudFront instead of S3 would make it much easier, but I can open a separate issue for that. My point is that we already have a URL identifying where to get a package from, so adding a second URL and a URL template that catalogs are required to support adds unnecessary complexity and out of band information to the package catalog protocol.
